### PR TITLE
BlockingQueue and guard adding null

### PR DESCRIPTION
**Pull request by tbradellis**
PR for:
https://github.com/newrelic/newrelic-java-agent/issues/621

We intermittently see NPEs from:
at com.newrelic.agent.stats.StatsServiceImpl.doStatsWork(StatsServiceImpl.java:61) ~[newrelic.jar:7.4.2]
at com.newrelic.agent.service.ServiceManagerImpl.replayStartupStatsWork(ServiceManagerImpl.java:585)
Which is preventing application startup with the agent.
They get the following error in newrelic.log when that happens. 

```
java.lang.NullPointerException: null
at com.newrelic.agent.stats.StatsServiceImpl.doStatsWork(StatsServiceImpl.java:61) ~[newrelic.jar:7.4.2]
at com.newrelic.agent.service.ServiceManagerImpl.replayStartupStatsWork(ServiceManagerImpl.java:585) ~[newrelic.jar:7.4.2]
at com.newrelic.agent.service.ServiceManagerImpl.doStart(ServiceManagerImpl.java:195) ~[newrelic.jar:7.4.2]
at com.newrelic.agent.service.AbstractService.start(AbstractService.java:63) ~[newrelic.jar:7.4.2]
at com.newrelic.agent.Agent.continuePremain(Agent.java:162) [newrelic.jar:7.4.2]
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_171]
```

We have been unable to replicate the issue. However there is general agreement that we can and should just guard against the NPE in that spot. It's also possible that another thread causes an issue. 

This PR changes the List holding StatsWork to a LinkedBlockingQueue and guards against attempting to add a null to the queue (which would throw an NPE for LBQ).  Logs that this occurred.

### CODE REVIEW

File: newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/PointCutClassTransformer.java [hunk 1]
```diff
@@ -381,7 +381,8 @@ public WeavingLoaderImpl(ClassLoader classLoader) {
         private void recordSupportabilityMetrics(List<PointCut> appliedPointCuts) {
             for (PointCut pointCut: appliedPointCuts) {
                 ServiceFactory.getStatsService().doStatsWork(StatsWorks.getRecordMetricWork(MessageFormat.format(
-                        MetricNames.SUPPORTABILITY_POINTCUT_LOADED, pointCut.getName()), 1));
+                        MetricNames.SUPPORTABILITY_POINTCUT_LOADED, pointCut.getName()), 1), MetricNames.SUPPORTABILITY_POINTCUT_LOADED +
+                        " " + pointCut.getName());
```
Comment (meiao) [inline newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/PointCutClassTransformer.java:385]: The log message for this entry will look odd:
Supportability/PointCutInstrumentation/Loaded/{0} pointcutName


As a general rule, when the metric name is created using String/Message.format, so should the statsWorkName.
Easiest way to do this is to create a var with the metric name and using it again as the statsWorkName.

File: newrelic-agent/src/main/java/com/newrelic/agent/service/ServiceManagerImpl.java [hunk 2]
```diff
@@ -582,15 +607,40 @@ public CircuitBreakerService getCircuitBreakerService() {
 
     private void replayStartupStatsWork() {
         for (StatsWork work : statsWork) {
-            statsService.doStatsWork(work);
+            //TODO: Figure out the best thing to do here. By this point, there is no way to have a null because LinkedBlockingQueue doesn't allow them and the queue is guarded against it anyway.
```
Comment (meiao) [inline newrelic-agent/src/main/java/com/newrelic/agent/service/ServiceManagerImpl.java:610]: I think we can remove this todo

File: newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/PointCutClassTransformer.java (hunk 1 above)
Comment (tbradellis) [reply newrelic-agent/src/main/java/com/newrelic/agent/instrumentation/PointCutClassTransformer.java]: done

File: newrelic-agent/src/main/java/com/newrelic/agent/service/ServiceManagerImpl.java (hunk 2 above)
Comment (tbradellis) [reply newrelic-agent/src/main/java/com/newrelic/agent/service/ServiceManagerImpl.java]: done

**Review by meiao — APPROVED** [review]

**Pull request merged**